xcode/Subconscious/Shared/Components/FirstRun/FirstRunView.swift

```diff
@@ -46,55 +72,69 @@ struct FirstRunView: View {
                 
                 Spacer()
                 
-                ValidatedFormField(
-                    alignment: .center,
-                    placeholder: "Enter your invite code",
-                    field: app.state.inviteCodeFormField,
-                    send: Address.forward(
-                        send: app.send,
-                        tag: AppAction.inviteCodeFormField
-                    ),
-                    caption: "Look for this in your welcome email.",
-                    submitLabel: .go,
-                    onSubmit: {
-                        if app.state.inviteCodeFormField.isValid {
-                            app.send(.pushFirstRunStep(.nickname))
+                if let gatewayId = app.state.gatewayId {
+                    InviteCodeRedeemedView(
+                        gatewayId: gatewayId
+                    )
+                } else {
+                    ValidatedFormField(
+                        alignment: .center,
+                        placeholder: "Enter your invite code",
+                        field: app.state.inviteCodeFormField,
+                        send: Address.forward(
+                            send: app.send,
+                            tag: AppAction.inviteCodeFormField
+                        ),
+                        caption: Func.run {
+                            switch app.state.inviteCodeRedemptionStatus {
+                            case .failed(_):
+                                return "Could not redeem invite code"
+                            case _:
+                                return "Your ticket to the Noosphere"
```

Comment:
I tweaked this to address #737 but open to changing to anything else

Reply:
I thought that a note for where to look was a good idea for the FRE in particular.

Perhaps "You can find your invite code in your welcome email"?